Suspend/Restore event policies around menu's place commands.

shaoshing opened the pull request:
This is similar to libraries drag-n-drop, the event policies have to be suspended before executing place commands via the menu item, and then restored afterward; otherwise, DS will be freezed. Addresses issue #2770

iwehrman commented on src/js/actions/menu.js:
```diff
@@ -63,18 +74,38 @@ define(function (require, exports) {
+                }
+            })
+            .catch(function () {
+                if (!isPlaceCommand) {
```
I'm confused as to why this happens when it's _not_ a place command that fails?

iwehrman replied:
Even if this is what you want, can you leave a comment here explaining why there is a failure and why it's OK to catch it and what the transfer accomplishes?

shaoshing replied:
We suspend the policies before running the place command, so here we want to make sure they are restored if any exceptions raised. I also added comment to explain to reason.

iwehrman replied:
Yes, but this says `if (!placeCommand)`. I understand that we want to clean up after suspending policies as a result of  place command. But it seems like this will do the cleanup only after executing something _other than_ a place command. Sorry if I'm just not getting it :)

iwehrman commented on src/js/actions/menu.js:
```diff
@@ -63,18 +74,38 @@ define(function (require, exports) {
+        return Promise.bind(this)
+            .then(function () {
+                if (isPlaceCommand) {
+                    this.dispatchAsync(events.menus.PLACE_COMMAND, { executing: true });
```
You probably want to just use `this.dispatch` here? Otherwise, the dispatch may not be complete by the time your transfer command executes.

shaoshing commented:
Hi @iwehrman , thanks for reviewing; I've addressed your comments. I also made the change to ease issue #2836 so that placing graphic from cc libraries won't steal all the policies. Right now we are waiting the core to fix the other part of the issue. Back to you.

shaoshing commented:
@iwehrman I've fixed the the obvious bug you pointed out; I appreciate for double checking it and did not let it slip thought :D

iwehrman commented:
OK, let's give it a whirl!
